tests/phpunit/SnowflakeTest.php

```diff
@@ -57,26 +60,32 @@ public function testDefaultWarehouse(): void
 
         $this->setUserDefaultWarehouse($user);
 
-        // run without warehouse param
-        unset($config['parameters']['db']['warehouse']);
-        $app = new SnowflakeApplication($config, new Logger(), [], $this->dataDir);
-
         try {
-            $app->run();
-            $this->fail('Run extractor without warehouse should fail');
-        } catch (Throwable $e) {
-            $this->assertMatchesRegularExpression('/No active warehouse/ui', $e->getMessage());
-        }
-
-        // run with warehouse param
-        $config = $this->getConfig();
-        $app = new SnowflakeApplication($config, new Logger(), [], $this->dataDir);
+            // run without warehouse param
+            unset($config['parameters']['db']['warehouse']);
+            $app = new SnowflakeApplication($config, new Logger(), [], $this->dataDir);
+
+            try {
+                $app->run();
+                $this->fail('Run extractor without warehouse should fail');
+            } catch (UserExceptionInterface $e) {
+                $this->assertSame(
+                    'Error connecting to DB: ' .
+                    'Please configure "warehouse" parameter. User default warehouse is not defined.',
+                    $e->getMessage()
+                );
+            }
 
-        $result = $app->run();
-        $this->assertEquals('success', $result['status']);
-        $this->assertCount(3, $result['imported']);
+            // run with warehouse param
+            $config = $this->getConfig();
+            $app = new SnowflakeApplication($config, new Logger(), [], $this->dataDir);
 
-        $this->setUserDefaultWarehouse($user, $warehouse);
+            $result = $app->run();
+            $this->assertEquals('success', $result['status']);
+            $this->assertCount(3, $result['imported']);
+        } finally {
+            $this->setUserDefaultWarehouse($user, $warehouse);
```

Comment:
Dal som to cele do try + finally, pretoze ak spadol test, tak moj testovaci user zostal bez default warehouse a padali ine testy.